include/deal.II/fe/fe_series.h

```diff
@@ -96,24 +96,29 @@ namespace FESeries
      * each direction, @p fe_collection is the hp::FECollection for which
      * expansion will be used and @p q_collection is the hp::QCollection used to
      * integrate the expansion for each FiniteElement in @p fe_collection.
+     * @p component defines which component of each FiniteElement will be used,
+     * it is mainly used for non-primitive elements.
      */
     Fourier(const std::vector<unsigned int> &      n_coefficients_per_direction,
             const hp::FECollection<dim, spacedim> &fe_collection,
-            const hp::QCollection<dim> &           q_collection);
+            const hp::QCollection<dim> &           q_collection,
+            const unsigned int                     component = 0);
```

Comment:
Here and other occasions: I would suggest to use `numbers::invalid_unsigned_int` as a default parameter.

If users provide vector-valued FEs and use this function, this function will work only on the very first component -- unknowing to the user. Instead, with the invalid parameter, you can check in the constructor with an assertion if the provided `fe_collection` consists of scalar or vector-valued finite elements, and then set `component` to zero if it only contains scalar FEs. This way users will be alarmed whether they are doing something unintentionally.

Reply:
Yes, I agree.